pkg/sync/sync_context.go

```diff
@@ -513,6 +513,17 @@ func (sc *syncContext) Sync() {
 	}
 }
 
+// filter out out-of-sync tasks
+func (sc *syncContext) filterOutOfSyncTasks(tasks syncTasks) syncTasks {
+	return tasks.Filter(func(t *syncTask) bool {
+		if modified, ok := sc.modificationResult[t.resourceKey()]; !modified && ok && t.targetObj != nil && t.liveObj != nil {
```

Comment:
Found another issue while testing the change. Sync hooks with hard-coded names should not filtered out. It is really easy to fix:

```golang
func (sc *syncContext) filterOutOfSyncTasks(tasks syncTasks) syncTasks {
	return tasks.Filter(func(t *syncTask) bool {
		if t.isHook() {
			return true
		}
		if modified, ok := sc.modificationResult[t.resourceKey()]; !modified && ok && t.targetObj != nil && t.liveObj != nil {
			sc.log.WithValues("resource key", t.resourceKey()).V(1).Info("Skipping as resource was not modified")
			return false
		}
		return true
	})
}
```